### ISSUE TITLE
Ending `poetry init` with Ctrl + D unexpectedly terminates whole `init` wizard

### ISSUE THREAD

**Issue by nchammas** [sdispater/poetry#309]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS 10.13.5
- **Poetry version**: 0.11.2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: this error comes up while trying to generate a pyproject.toml file with Poetry

## Issue

Here's the reproduction:

```sh
$ poetry -vvv init
<initial setup snipped>
Search for package: boto3

Found 47 packages matching boto3

Enter package # to add, or the complete package name if it is not listed: 
 [ 0] Flask-Boto3
 [ 1] boto3
<more options snipped>
 > 1
stty: 'standard input': unable to perform all requested operations

Enter the version constraint to require (or leave blank to use the latest version): 
Using version ^1.7 for boto3

Search for a package: <Ctrl + D here>
[RuntimeError]  
Aborted                      
                               
Exception trace:
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /usr/local/lib/python3.6/site-packages/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /usr/local/lib/python3.6/site-packages/poetry/console/commands/command.py in run() at line 72
   return super(BaseCommand, self).run(i, o)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /usr/local/lib/python3.6/site-packages/poetry/console/commands/init.py in handle() at line 114
   self._determine_requirements(self.option("dependency"))
 /usr/local/lib/python3.6/site-packages/poetry/console/commands/init.py in _determine_requirements() at line 205
   package = self.ask("\nSearch for a package:")
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/commands/command.py in ask() at line 213
   return self.output.ask(question, default)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/styles/cleo_style.py in ask() at line 157
   return self.ask_question(question)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/styles/cleo_style.py in ask_question() at line 189
   answer = QuestionHelper().ask(self._input, self, question)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/helpers/question_helper.py in ask() at line 49
   return self._do_ask(output, question)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/helpers/question_helper.py in _do_ask() at line 91
   ret = self._read_from_input(input_stream)
 /usr/local/lib/python3.6/site-packages/poetry/_vendor/cleo/helpers/question_helper.py in _read_from_input() at line 350
   raise RuntimeError('Aborted')

init [--name NAME] [--description DESCRIPTION] [--author AUTHOR] [--dependency DEPENDENCY] [--dev-dependency DEV-DEPENDENCY] [-l|--license LICENSE]
```

Basically, it's a combination of two minor issues.

First, it's not clear how to terminate the repeated prompts of `Search for a package:`. Perhaps a hint like "Leave blank and hit Enter to move to next step" would make it more obvious.

Since I didn't know at first what to do to move to the next stage, I tried Ctrl + D, which is commonly used to mean "end of input". I guess that was the wrong thing here, since it terminated the whole initialization wizard and I had to start over.

I think a small tweak or two to `poetry init` could help others avoid a similar situation in the future.

**Comment by stephsamson** [comment]
Hey Nick! Great to see you here on the issue tracker in `poetry` :)

#1291 fixes this issue so I'll close this for now. If you think that other improvments could be made, feel free to re-open this or create a new one.
